JAR for 12.1.1 is missing org.slf4j package

The JAR for checkstyle 12.1.1 seems to be missing the `org.slf4j` package which causes it to raise an exception when attempting to run it with the `sarif` output format:

```bash
/code # java -jar /opt/checkstyle/checkstyle.jar -f sarif -c /sun_checks.xml .
Exception in thread "main" java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
	at org.reflections.Reflections.<clinit>(Reflections.java:115)
	at com.puppycrawl.tools.checkstyle.meta.XmlMetaReader.readAllModulesIncludingThirdPartyIfAny(XmlMetaReader.java:75)
	at com.puppycrawl.tools.checkstyle.SarifLogger.loadModuleMetadata(SarifLogger.java:212)
	at com.puppycrawl.tools.checkstyle.SarifLogger.<init>(SarifLogger.java:188)
	at com.puppycrawl.tools.checkstyle.Main$OutputFormat.createListener(Main.java:588)
	at com.puppycrawl.tools.checkstyle.Main.createListener(Main.java:516)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:421)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:347)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:206)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 10 more
```
---

Checkstyle 12.1.0:

<img width="820" height="394" alt="Image" src="https://github.com/user-attachments/assets/59024e03-e4a5-4e82-8c76-b90b4354e5d3" />

Checkstyle 12.1.1:

<img width="790" height="372" alt="Image" src="https://github.com/user-attachments/assets/52ffbc83-876b-497b-be08-6ede0649a546" />

--- a/.ci/validation.sh
+++ b/.ci/validation.sh
@@ -450,17 +450,51 @@ assembly-run-all-jar)
   mkdir -p .ci-temp
   FOLDER=src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired
   FILE=InputMissingJavadocTypeCorrect.java
+  echo "Execution with plain text report"
   java -jar target/checkstyle-"$CS_POM_VERSION"-all.jar -c /google_checks.xml \
         $FOLDER/$FILE > .ci-temp/output.log
   fail=0
   if grep -vE '(Starting audit)|(warning)|(Audit done.)' .ci-temp/output.log ; then
     fail=1;
+    exit $fail;
   elif grep 'warning' .ci-temp/output.log ; then
     fail=1;
+    exit $fail;
   fi
   rm .ci-temp/output.log
-  sleep 5
-  exit $fail
+  echo "Execution with xml report"
+  java -jar target/checkstyle-"$CS_POM_VERSION"-all.jar -f xml -c /google_checks.xml \
+        $FOLDER/$FILE -o .ci-temp/output.xml
+  fail=0
+  echo "Content of report:"
+  cat .ci-temp/output.xml
+  echo "Validation of report"
+  if ! grep '</checkstyle>' .ci-temp/output.xml ; then
+    fail=1;
+    echo "no closed tag"
+    exit $fail;
+  elif ! grep 'file name' .ci-temp/output.xml ; then
+    fail=1;
+    echo "no file tag"
+    exit $fail;
+  fi
+  rm .ci-temp/output.xml
+  echo "Execution with sarif report"
+  java -jar target/checkstyle-"$CS_POM_VERSION"-all.jar -f sarif -c /google_checks.xml \
+        $FOLDER/$FILE -o .ci-temp/output.json
+  fail=0
+  echo "Content of report:"
+  cat .ci-temp/output.json
+  echo "Validation of report"
+  if ! grep 'downloadUri' .ci-temp/output.json ; then
+    fail=1;
+    exit $fail;
+  elif ! grep 'results' .ci-temp/output.json ; then
+    fail=1;
+    exit $fail;
+  fi
+  rm .ci-temp/output.json
+
   ;;
 
 check-since-version)

--- a/pom.xml
+++ b/pom.xml
@@ -208,7 +208,7 @@
     <pmd.version>7.18.0</pmd.version>
     <maven.jacoco.plugin.version>0.8.14</maven.jacoco.plugin.version>
     <mockito.version>5.2.0</mockito.version>
-    <slf4j.version>2.0.17</slf4j.version>
+    <slf4j.version>1.7.32</slf4j.version> <!-- Aligned with the version reflections depends on -->
     <saxon.version>12.9</saxon.version>
     <maven.checkstyle.plugin.version>3.6.0</maven.checkstyle.plugin.version>
     <maven.sevntu.checkstyle.plugin.version>1.44.1</maven.sevntu.checkstyle.plugin.version>
@@ -426,13 +426,13 @@
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
       <version>${slf4j.version}</version>
-      <scope>test</scope>
+      <scope>runtime</scope>
     </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-simple</artifactId>
       <version>${slf4j.version}</version>
-      <scope>test</scope>
+      <scope>runtime</scope>
     </dependency>
     <dependency>
       <groupId>org.jacoco</groupId>

